src/main/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisObserver.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.gridsuite.sensitivityanalysis.server.service;
+
+import com.powsybl.sensitivity.SensitivityAnalysisResult;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.observation.Observation;
+import io.micrometer.observation.ObservationRegistry;
+import lombok.NonNull;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+/**
+ * @author Florent MILLOT <florent.millot at rte-france.com>
+ */
+@Service
+public class SensitivityAnalysisObserver {
+
+    private static final String OBSERVATION_PREFIX = "app.computation.";
+    private static final String PROVIDER_TAG_NAME = "provider";
+    private static final String TYPE_TAG_NAME = "type";
+    private static final String STATUS_TAG_NAME = "status";
+    private static final String COMPUTATION_TYPE = "sensi";
+    private static final String COMPUTATION_COUNTER_NAME = OBSERVATION_PREFIX + "count";
+
+    private final String defaultProvider;
+    private final ObservationRegistry observationRegistry;
+    private final MeterRegistry meterRegistry;
+
+    public SensitivityAnalysisObserver(@Value("${sensitivity-analysis.default-provider}") String defaultProvider,
+                                       @NonNull ObservationRegistry observationRegistry,
+                                       @NonNull MeterRegistry meterRegistry) {
+        this.defaultProvider = defaultProvider;
+        this.observationRegistry = observationRegistry;
+        this.meterRegistry = meterRegistry;
+    }
+
+    public <E extends Throwable> void observe(String name, SensitivityAnalysisRunContext runContext, Observation.CheckedRunnable<E> callable) throws E {
+        createObservation(name, runContext).observeChecked(callable);
```

Comment:
```suggestion
    public <E extends Throwable> void observe(String name, SensitivityAnalysisRunContext runContext, Observation.CheckedRunnable<E> runnable) throws E {
        createObservation(name, runContext).observeChecked(runnable);
```

Reply:
I just copy paste from other MS https://github.com/gridsuite/security-analysis-server/blob/6bab1554756090fef76bf5ec82dd3d6600fb4fd5/src/main/java/org/gridsuite/securityanalysis/server/service/SecurityAnalysisObserver.java#L44-L46
But indeed it should be changed everywhere.

Reply:
https://github.com/gridsuite/security-analysis-server/pull/106